accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,212 @@
+        'BLOCK_ACTION_LIST': Blockly.Msg.BLOCK_ACTION_LIST,
+        'PASTE': Blockly.Msg.PASTE,
+        'PASTE_ABOVE': Blockly.Msg.PASTE_ABOVE,
+        'PASTE_BELOW': Blockly.Msg.PASTE_BELOW,
+        'MARK_THIS_SPOT': Blockly.Msg.MARK_THIS_SPOT,
+        'MARK_SPOT_ABOVE': Blockly.Msg.MARK_SPOT_ABOVE,
+        'MARK_SPOT_BELOW': Blockly.Msg.MARK_SPOT_BELOW,
+        'CUT_BLOCK': Blockly.Msg.CUT_BLOCK,
+        'COPY_BLOCK': Blockly.Msg.COPY_BLOCK,
+        'MOVE_TO_MARKED_SPOT': Blockly.Msg.MOVE_TO_MARKED_SPOT,
+        'DELETE': Blockly.Msg.DELETE,
+      };
+    }],
+    ngOnInit: function(){
+      var elementsNeedingIds = ['blockSummary', 'listItem', 'label',
+          'cutListItem', 'cutButton', 'copyListItem', 'copyButton',
+          'pasteBelow', 'pasteBelowButton', 'pasteAbove', 'pasteAboveButton',
+          'markBelow', 'markBelowButton', 'markAbove', 'markAboveButton',
+          'sendToSelectedListItem', 'sendToSelectedButton', 'delete',
+          'deleteButton'];
+      for (var i=0; i<this.block.inputList.length; i++){
+        var inputBlock = this.block.inputList[i];
+        if (inputBlock.connection && !inputBlock.connection.targetBlock()){
+          elementsNeedingIds.concat(['inputList' + i, 'inputMenuLabel' + i, 'markSpot' + i,
+             'markSpotButton' + i, 'paste' + i, 'pasteButton' + i]);
+        }
+      }
+      this.idMap = this.utilsService.generateIds(elementsNeedingIds);
+      this.idMap['parentList'] = this.generateParentListId();
+    },
+    generateParentListId: function(){
+      if (this.isTopBlock){
+        return this.parentId + '-node0'
+      } else {
+        return this.utilsService.generateUniqueId();
+      }
+    },
+    noPreviousConnectionHTMLText: function(block) {
+      if (!block.previousConnection) {
+        return 'blockly-disabled';
+      } else {
+        return '';
+      }
+    },
+    noNextConnectionHTMLText: function(block) {
+      if (!block.nextConnection) {
+        return 'blockly-disabled';
+      } else {
+        return '';
+      }
+    },
+    checkParentList: function(parentList) {
+      blocklyApp.debug && console.log('setting parent list');
+      var tree = parentList;
+      var regex = /^blockly-workspace-tree\d+$/;
+      while (tree && !tree.id.match(regex)) {
+        tree = tree.parentNode;
+      }
+      if (tree && tree.getAttribute('aria-activedescendant') == parentList.id) {
+        this.treeService.updateSelectedNode(parentList, tree, false);
+      }
+    },
+    setId: function(block) {
+      if (this.isTopBlock) {
+        return this.parentId + '-node0';
+      }
+      return this.treeService.createId(block);
+    },
+    sendToSelected: function(block) {
+      if (this.clipboardService) {
+        this.clipboardService.pasteToMarkedConnection(block, false);
+        block.dispose(true);
+        alert('Block moved to marked spot: ' + block.toString());
+      }
+    },
+    clipboardCompatibilityHTMLText: function(connection) {
```

Comment:
Those functions in the clipboardService return booleans. I think you're thinking of some of the HTMLText functions that are in the utilsService? This function didn't make it there because it's not used by any other files.